Firestore/core/src/local/leveldb_key.cc

```diff
@@ -1346,10 +1348,75 @@ bool LevelDbDocumentOverlayLargestBatchIdIndexKey::Decode(
     absl::string_view key) {
   Reader reader{key};
   reader.ReadTableNameMatching(kDocumentOverlaysLargestBatchIdIndexTable);
-  user_id_ = reader.ReadUserId();
-  largest_batch_id_ = reader.ReadBatchId();
-  document_key_ = reader.ReadDocumentKey();
+  auto user_id = reader.ReadUserId();
+  auto largest_batch_id = reader.ReadBatchId();
+  auto document_key = reader.ReadDocumentKey();
+  reader.ReadTerminator();
+  Reset(std::move(user_id), std::move(largest_batch_id),
+        std::move(document_key));
+  return reader.ok();
+}
+
+std::string LevelDbDocumentOverlayCollectionIndexKey::KeyPrefix() {
```

Comment:
This is not used outside of tests, but we could keep it for consistency.

Reply:
... and even the usages of `KeyPrefix()` in tests was only to verify its behavior. I've removed these unnecessary `KeyPrefix()` methods.